### PR TITLE
[MRG] Auto rename duplicate simulations

### LINKED ISSUE #1137: ENH/GUI: Auto-append new simulation name in GUI
In the GUI, when the user tries to run a second simulation, but they have not explicitly changed the name of the simulation, we should change the behavior. Currently, there is a hard stop and the user is forced to change the name. It would be more user-friendly if, instead, in this situation we simply changed the name automatically and appended some text to it, such as `default` becoming `default-002`, etc. This way, the user can always explicitly change the name if they want to, but they don't have to, especially if they're iterating between values in one of the parameter tabs.

### PR DESCRIPTION
**Pull request by Tusharjamdade**
Fixes #1137 
Improves the GUI simulation workflow by removing the hard stop when a user reruns a simulation without changing its name. Instead of forcing a manual rename, the GUI now automatically generates a unique simulation name by appending an incremented numeric suffix.
```
default → default-2 → default-3 → default-4 → …
```

### CODE REVIEW

File: hnn_core/gui/gui.py [hunk 1]
```diff
@@ -2686,9 +2686,13 @@ def run_button_clicked(
...
         if simulation_data[_sim_name]["net"] is not None:
-            print("Simulation with the same name exists!")
-            simulation_status_bar.value = simulation_status_contents["failed"]
-            return
+            base = _sim_name.split("-")[0]
+            idx = 2
+            while f"{base}-{idx:03d}" in simulation_data:
+                idx += 1
+
+            _sim_name = f"{base}-{idx:03d}"
```
Comment (ntolley) [inline hnn_core/gui/gui.py]:
```suggestion
            _sim_name = f"{base}-{idx}"
```
I think it's probably fine to not use fixed with number formatting

**Comment by Tusharjamdade** [comment]
Hi @ntolley, thank you for the review! I’ve made the suggested changes, please take a look when you get a chance.

**Comment by Tusharjamdade** [comment]
Hi maintainers, this PR is for GSoC 2026. Could you please add the `gsoc-2026` label? I would also appreciate a review whenever you have time. Thank you!

**Comment by Tusharjamdade** [comment]
Thank you for the detailed review! I've updated the implementation to use `rsplit("-", 1)` so that hyphenated names are preserved and existing numeric suffixes are incremented correctly (e.g., sim-34 → sim-35 instead of sim-34-2).

The logic now handles all the edge cases you pointed out, and the new tests are passing. Please let me know if you'd like any further refinements.
